### DISCUSSION TITLE
SEQ API HTTP (Seq.API) read @Message of Exception log

### DISCUSSION THREAD

**Discussion by mKamiAsim** [datalust/seq-tickets#1962] [Help] [answered]
I want to read @Message of Exception by SeqConnection.Events.ListAsync(). Exception details are available in EventEntity.Exception with stack trace. I want to get only Exception message. Is there any way to get Message only (Not cropping message from Exception property)

@Message can be easily extracted from below seq query. but unable to find such property in SEQ API HTTP CLIENT

Select @Message from stream where @Level in ['Error','Fatal'] limit 10

**Comment by nblumhardt** [comment, marked as answer]
Hi @mKamiAsim,

The `@Message` property on the server is separate from the message part of the exception (which is contained within the `@Exception` string).

Because of the design of .NET exceptions, there isn't always only a single message within an exception, because of wrapper exception types like `AggregateException`.

If your use case relies heavily on having the message stored alongside the events, using a Serilog `ILogEventEnricher` in your applications to attach a separate `ExceptionMessage` property will be the most reliable long-term.

Otherwise, pulling the message out of the full exception string using a regular expression is the only way to achieve this.

Hope this helps,
Nick
